### DISCUSSION TITLE
Change PDF Template

### DISCUSSION THREAD

**Discussion by freelowdr** [SolidInvoice/SolidInvoice#955] [General]
Hi, I'd like to make some changes to how the pdf looks so that it generates according to my preferences. I assume there is a template file or something it uses to determine this but I tested several files and am unable to find the right one.
Which file(s) do I edit to make desired changes such as where data is displayed, formatting and logo placement?
Worth noting I'm aware that it is not persistent in the docker container therefore I'll utilize bind mounts where necessary. Any assistance would be greatly appreciated. I'm sure there are others that will be interested in this functionality. I'm really enjoying solidinvoice, and it works very well, thanks.

**Comment by pierredup** [comment]
@freelowdr The templates for the PDF is available at `src/InvoiceBundle/Resources/views/Pdf/invoice.html.twig` for invoices and `src/QuoteBundle/Resources/views/Pdf/quote.html.twig` for quotes.

If you make any changes to these files, then you would need to clear the cache since the templates are cached (this can be done by deleting the `var/cache/prod/twig` directory).

We would also welcome any contributions to improve the PDF templates or provide new templates, so if you have any changes that you think might benefit other users, please suggest the changes in an issue and we will gladly incorporate it into SolidInvoice.

Thanks for the support!

**Reply by bilogic** [reply]
@pierredup 

Any plans to let user add their own dynamic QR codes (codes with embedded unique meta data, e.g. amount to pay/reference # etc) to support the different kinds of scan to pay QR codes?

**Reply by pierredup** [reply]
@bilogic How would you like to embed the QR code? Would you generate it in a separate system, and download an image of the QR code to be added to the invoice?
	
This would be possible, but there would then be no callback to notify SolidInvoice that the invoice is paid (I.E when the client pays with the QR code, the invoice would still be marked as Pending in SolidInvoice and would manually need to be marked as Paid)

**Reply by bilogic** [reply]
Yea it will have to be a step by step thing. I don't a decided way to embed yet but perhaps a generic and easy way is to allow us to specify a URL to call to download the QR code. The URL will have to be configurable with the invoice variables though. Make sense?

**Comment by Luzl1** [comment]
Sorry for resurrecting this thread ...
I'm trying to replace the standard invoice template. As @pierredup writes this is in `src/InvoiceBundle/Resources/views/Pdf/invoice.html.twig` but I can't find the tempalte inside the container.
Am I right, that I have to build the container image by myself with the changed template?
